src/JsonLdWriter.php

```diff
@@ -288,7 +288,7 @@ protected function convertImageToBinary(array|string $value): mixed
                 // Some other form of image like a Barcode and ImageObjectSnapshot.
                 // See: https://schema.org/ImageObject
                 $this->logger->warning(
-                    "Images of type ". $value['@type']
+                    "Images of type ".  json_encode($value['@type'])
```

Comment:
The original code directly used $value['@type'] within a string concatenation without checking its type. If $value['@type'] was an array instead of a string, it would cause an "Array to string conversion" error. For this, checks can be added to ensure that $value['@type'] is not empty before encoding it, in case it is an array.

Reply:
Just synced with @pujak17 : The underlying issue here is that actually `@type` can be empty in the source data, which results to `@type` being set to an empty array in the PHP object.

Likely the best solution here is to throw a new additional warning that states that `@type` is empty.

Reply:
While you are at it try to invest some 5 min to find out where a good place could be for setting a logger (instead of current NullLogger).

Reply:
Also the merge conflict has also been fxed